pkg/controller/certificates/signer/signer.go

```diff
@@ -44,56 +46,37 @@ func NewCSRSigningController(
 	if err != nil {
 		return nil, err
 	}
+
 	return certificates.NewCertificateController(
 		"csrsigning",
 		client,
 		csrInformer,
 		signer.handle,
-	), nil
+	).WithDynamicCertReloader(signer.caLoader), nil
 }
 
 type signer struct {
-	ca      *authority.CertificateAuthority
+	caValue  atomic.Value
+	caLoader *dynamiccertificates.DynamicFileServingContent
+
 	client  clientset.Interface
 	certTTL time.Duration
 }
 
 func newSigner(caFile, caKeyFile string, client clientset.Interface, certificateDuration time.Duration) (*signer, error) {
-	certPEM, err := ioutil.ReadFile(caFile)
+	caLoader, err := dynamiccertificates.NewDynamicServingContentFromFiles("csr-controller", caFile, caKeyFile)
```

Comment:
"Serving" in NewDynamicServingContentFromFiles is not accurate in this context. It's probably worth renaming.